src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -119,24 +123,29 @@ export default class GeoChartRenderer extends React.Component<IGeoChartRendererP
             container: this.chartRef,
             // If true, the map’s canvas can be exported to a PNG using map.getCanvas().toDataURL().
             // This is false by default as a performance optimization.
-            preserveDrawingBuffer: isExportMode,
             interactive: !isViewportFreezed,
+            preserveDrawingBuffer: this.isExportMode,
         });
     };
 
     public render() {
-        const isExportMode = this.isExportMode();
+        const {
+            config: { groupNearbyPoints, isExportMode = false },
+            geoData,
+        } = this.props;
+        this.isExportMode = isExportMode;
+        this.isClusteringAllowed = isClusteringAllowed(geoData, groupNearbyPoints);
```

Comment:
We should not make var assignment in `render` lifecycle.
It should be in componentDidUpdate and/or componentDidMount.

Besides, I'm wondering why we persist them. They can be calculated where they are used, can't they?